src/TimeModeling/Types/abstract.jl

```diff
@@ -73,7 +73,15 @@ vec(x::judiMultiSourceVector) = vcat(vec.(x.data)...)
 
 time_sampling(ms::judiMultiSourceVector) = [1 for i=1:ms.nsrc]
 
-reshape(ms::judiMultiSourceVector, dims::Dims{N}) where N = reshape(vec(ms), dims)
+function reshape(ms::judiMultiSourceVector, dims::Dims{N}) where N
```

Comment:
This is for

<details>

<summary>failing example</summary>

```julia
using JUDI
using Flux
using ArgParse, Test, Printf, Aqua
using SegyIO, LinearAlgebra, Distributed, JOLI
using TimerOutputs: TimerOutputs, @timeit

Flux.Random.seed!(2022)

### Model
tti = false
viscoacoustic = false

nsrc = 1
dt = 1f0
include(joinpath(JUDIPATH, "../test/seismic_utils.jl"))
model, model0, dm = setup_model(tti, viscoacoustic, 4)
m, m0 = model.m.data, model0.m.data
q, srcGeometry, recGeometry, f0 = setup_geom(model; nsrc=nsrc, dt=dt)

# Common op
Pr = judiProjection(recGeometry)
Ps = judiProjection(srcGeometry)

opt = Options(sum_padding=true, f0=f0)
A_inv = judiModeling(model; options=opt)
A_inv0 = judiModeling(model0; options=opt)

# Operators
F = Pr*A_inv*adjoint(Ps)
J = judiJacobian(F,q)
dm = vec(m-m0)

gs_inv = gradient(q -> norm(J(q)*dm), q)
```

```julia
ERROR: LoadError: DimensionMismatch: new dimensions (1,) must be consistent with array size 1501
Stacktrace:
  [1] (::Base.var"#throw_dmrsa#289")(dims::Tuple{Int64}, len::Int64)
    @ Base ./reshapedarray.jl:41
  [2] reshape
    @ ./reshapedarray.jl:45 [inlined]
  [3] reshape
    @ ~/.julia/dev/JUDI/src/TimeModeling/Types/abstract.jl:76 [inlined]
  [4] reshape(parent::judiVector{Float32, Matrix{Float32}}, shp::Tuple{Base.OneTo{Int64}})
    @ Base ./reshapedarray.jl:111
  [5] (::ChainRulesCore.ProjectTo{AbstractArray, NamedTuple{(:element, :axes), Tuple{ChainRulesCore.ProjectTo{Float32, NamedTuple{(), Tuple{}}}, Tuple{Base.OneTo{Int64}}}}})(dx::JUDI.LazyPropagation)
    @ JUDI ~/.julia/dev/JUDI/src/rrules.jl:142
  [6] _project
    @ ~/.julia/packages/Zygote/SmJK6/src/compiler/chainrules.jl:184 [inlined]
  [7] map(f::typeof(Zygote._project), t::Tuple{judiVector{Float32, Matrix{Float32}}}, s::Tuple{JUDI.LazyPropagation})
    @ Base ./tuple.jl:246
  [8] gradient(f::Function, args::judiVector{Float32, Matrix{Float32}})
    @ Zygote ~/.julia/packages/Zygote/SmJK6/src/compiler/interface.jl:98
  [9] top-level scope
    @ ~/.julia/dev/JUDI/test/MFE.jl:33
 [10] include(fname::String)
    @ Base.MainInclude ./client.jl:476
 [11] top-level scope
    @ REPL[1]:1
```

</details>

Reply:
It wasn't failing before what changed?

Reply:
The example above fails on master branch

Reply:
To be more specific:

```julia
julia> gs_inv = gradient(() -> norm(J(q)*dm), Flux.params(q))
Operator `born` ran in 0.75 s
Grads(...)
```

this doesn't fail but

```julia
julia> gs_inv = gradient(q -> norm(J(q)*dm), q)
Operator `born` ran in 0.72 s
Operator `born` ran in 0.73 s
ERROR: DimensionMismatch: new dimensions (1,) must be consistent with array size 1501
Stacktrace:
  [1] (::Base.var"#throw_dmrsa#289")(dims::Tuple{Int64}, len::Int64)
    @ Base ./reshapedarray.jl:41
  [2] reshape
    @ ./reshapedarray.jl:45 [inlined]
  [3] reshape
    @ ~/.julia/dev/JUDI/src/TimeModeling/Types/abstract.jl:76 [inlined]
  [4] reshape(parent::judiVector{Float32, Matrix{Float32}}, shp::Tuple{Base.OneTo{Int64}})
    @ Base ./reshapedarray.jl:111
  [5] (::ChainRulesCore.ProjectTo{AbstractArray, NamedTuple{(:element, :axes), Tuple{ChainRulesCore.ProjectTo{Float32, NamedTuple{(), Tuple{}}}, Tuple{Base.OneTo{Int64}}}}})(dx::JUDI.LazyPropagation)
    @ JUDI ~/.julia/dev/JUDI/src/rrules.jl:142
  [6] _project
    @ ~/.julia/packages/Zygote/SmJK6/src/compiler/chainrules.jl:184 [inlined]
  [7] map(f::typeof(Zygote._project), t::Tuple{judiVector{Float32, Matrix{Float32}}}, s::Tuple{JUDI.LazyPropagation})
    @ Base ./tuple.jl:246
  [8] gradient(f::Function, args::judiVector{Float32, Matrix{Float32}})
    @ Zygote ~/.julia/packages/Zygote/SmJK6/src/compiler/interface.jl:98
  [9] top-level scope
    @ REPL[25]:1
 [10] top-level scope
    @ ~/.julia/packages/CUDA/DfvRa/src/initialization.jl:52
```

this fail

Reply:
Hum ok, then split into `Dims{N}` and `Dims{1}` and just an if/else these try/catch are really a bad idea anywhere near Zygote